### PR TITLE
Ensure dbt crawler outputs unique entity IDs [sc-5857]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To fix the crawler output. Currently, the crawler could output duplicate entity ID in `sourceModels`, `sourceDatasets` or `owner` depending on the dbt `manifest.json`, which will break the uniqueness validation in Metaphor.

### 🤓 What?

- Uniquify entity ID preserving order.

### 🧪 Tested?

Uniquify `tests/dbt/data/shopify/results.json` manually and run tests successfully.

### CODE REVIEW

File: metaphor/dbt/extractor.py [hunk 1]
```diff
@@ -325,17 +325,21 @@ def _parse_manifest_nodes(
...
-                        for n in model.depends_on.nodes
-                        if n.startswith("model.")
-                    ]
-
-                    dbt_model.source_datasets = [
-                        str(source_map[n])
-                        for n in model.depends_on.nodes
-                        if n.startswith("source.")
-                    ]
+                    dbt_model.source_models = list(
```
Comment (mars-lan) [inline metaphor/dbt/extractor.py:328]: Consider encapsulating this in a utility function, e.g. `unique_list`, to improve readability.

**Comment by mars-lan** [comment]
Please also give a more description `Why`, e.g. "Currently the connector outputs duplicated keys in `sourceModels` which can break the uniqueness validation"

**Comment by mars-lan** [comment]
Please also bump up the version: https://github.com/MetaphorData/connectors/blob/main/pyproject.toml#L3

**Review by mars-lan — APPROVED** [review]
Great work!

File: metaphor/dbt/extractor.py [hunk 2]
```diff
@@ -480,3 +486,7 @@ def _init_field_doc(dataset: Dataset, column: str) -> FieldDocumentation:
             doc.field_path = column
             dataset.documentation.field_documentations.append(doc)
         return doc
+
+    @staticmethod
+    def _unique_list(vars: List[str]) -> List[str]:
+        return list(dict.fromkeys(vars))
```
Comment (alyiwang) [inline metaphor/dbt/extractor.py:492]: Sorry, late for the review. Maybe `list(set(values))` is simpler (more readable) ?

Comment (mars-lan) [reply metaphor/dbt/extractor.py:492]: I believe @elic-eon wants to preserve the order, which is not strictly required but a nice to have property.

Comment (elic-eon) [reply metaphor/dbt/extractor.py:492]: Yes, to prevent tests from randomly failing.

Comment (mars-lan) [reply metaphor/dbt/extractor.py:492]: oh yeah that's right I forgot about the tests.

**Pull request merged**